### DISCUSSION TITLE
friction and script based OrbitalNav

### DISCUSSION THREAD

**Discussion by hedbergj** [OpenSpace/OpenSpace#2097] [Q&A] [answered]
Is there a reason why the various friction (roll, rot, etc) settings do not apply when using `openspace.navigation.addGlobalRotation()` and other script activated motions? i.e. if I want to just add a global rotation that doesn't slow down and stop (until told to), just turning off friction doesn't seem to affect anything. The camera still just slows down and stops.

**Comment by micahnyc** [comment, marked as answer]
I feel like this is more of an oversight rather then an intended behavior. I'll see what other developers think and if so, we'll create an issue from this discussion.

**Reply by alexanderbock** [reply]
Yea, it's an oversight;  if someone can access the `addGlobalRotation` function, they can access the `setPropertyValue` function, too, to disable the fiction if they don't want it

**Reply by hedbergj** [reply]
You mean: `openspace.setPropertyValue("NavigationHandler.OrbitalNavigator.Friction.RotationalFriction", false)`? Isn't that the same as just un-checking the box in the gui or is there another command you're referencing? Basically, I want to enter: `openspace.navigation.addGlobalRotation(10,0);` and then just have it rotate in the given (and precise rather than mouse-messy) direction without stopping. My naive guess is that it has something to do with  `_scriptStates` not being listed here:

```
_friction.rotational.onChange([&]() {
        _mouseStates.setHorizontalFriction(_friction.rotational);
        _joystickStates.setHorizontalFriction(_friction.rotational);
        _websocketStates.setHorizontalFriction(_friction.rotational);
    });
 ```
 
 in `orbitalnavigator.cpp`

**Reply by alexanderbock** [reply]
Yea, I'm agreeing with Micah that it is an oversight that the `addGlobalRotation` doesn't listen to the friction value.  If you don't want friction, you should be able to:
```lua
openspace.setPropertyValue("NavigationHandler.OrbitalNavigator.Friction.RotationalFriction", false)
openspace.navigation.addGlobalRotation(10,0)
```
as you are saying :+1:

**Reply by hedbergj** [reply]
Isn't that functionally identical to just pressing `f` or unchecking the GUI box? (I tried the `setPropertyValue()` and it still comes to a halt.) Or do you mean, that's how _it should_ work?

**Reply by alexanderbock** [reply]
Yes, that's _how_ it should work not how it _does_ work

**Reply by alexanderbock** [reply]
Created an issue #2114 for it to be fixed

**Reply by hedbergj** [reply]
Hooray! Auxiliary Verbs FTW!
